### PR TITLE
Update regex date match patterns

### PR DESCRIPTION
**Pull request by ghost**
## Proposed change

This change should help fix missing detection of dates such as 20-FEB-2022. As far as I understand the existing regexes this is not yet implemented. I also added a simple example at the comment above the regexes. Since I'm new to GitHub I have a minimal understanding (this is my first PR), I hope I didn't make a mess.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other (please explain)

## Checklist:

- [x] I have read & agree with the [contributing guidelines](https://github.com/paperless-ngx/paperless-ngx/blob/main/CONTRIBUTING.md).
- [ ] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.
- [x] If applicable, I have checked that all tests pass, see [documentation](https://paperless-ngx.readthedocs.io/en/latest/extending.html#back-end-development).
- [x] I have run all `pre-commit` hooks, see [documentation](https://paperless-ngx.readthedocs.io/en/latest/extending.html#code-formatting-with-pre-commit-hooks).
- [x] I have made corresponding changes to the documentation as needed.
- [ ] I have checked my modifications for any breaking changes.

### CODE REVIEW

**Comment by stumpylog** [comment]
Could you add testing to confirm the new regex parsing works as intended?

**Comment by fantasticle** [comment]
I added a test function and it passed the test.
Interestingly it also passed the test when changing it to something that should fail. So I rechecked within the `python manage.py shell`.

```python
    from documents.parsers import parse_date
    from django.test import TestCase
    import re
    import datetime

    REGEX_DATE = re.compile(r"(\b|(?!=([_-])))([0-9]{1,2}[ \.\/-][A-Z]{3}[ \.\/-][0-9]{4})(\b|(?=([_-])))|")
    class MyDummyTest(TestCase):
        pass
    x = MyDummyTest()
    text = "Customer Number Currency 22-MAR-2022 Credit Card 1934829304"
    x.assertEqual(parse_date("",text), datetime.datetime(2020,1,1,0,0, tzinfo=tz.gettz(settings.TIME_ZONE)))
```
This fails like expected. Can you confirm?

**Comment by stumpylog** [comment]
Try: `r"(\b|(?!=([_-])))(\b[0-9]{1,2}[ \.\/-][A-Z]{3}[ \.\/-][0-9]{4})(\b|(?=([_-])))"` for the regex.  That passed the testing I could dream up:

-  "Customer Number Currency 22-MAR-2022 Credit Card 1934829304"
- "Something something 25/MAR-1925 Blah Numbers 551105112"
- "Something something 123-MAR-2022 Blah Numbers 123456789"  (no match expected)
- "Something something 25-MAR-23 Blah Numbers 987654321" (no match expected)
- "Something something 25-MAR-12345 Blah Numbers 1597536428" (no match expeted)

Then add back the trailing `,` so the checks will proceed to testing.

File: src/documents/tests/test_date_parsing.py [hunk 1]
```diff
@@ -92,6 +92,13 @@ def test_date_format_8(self):
             datetime.datetime(2020, 3, 1, 0, 0, tzinfo=tz.gettz(settings.TIME_ZONE)),
         )
 
+    def test_date_format_9(self):
```
[the remark this answers is not in the document]
Comment (fantasticle) [reply src/documents/tests/test_date_parsing.py:95]: Sure - I just thought to follow the basic theme of naming those test functions. Should I rename all of them and try to clarify what they are doing?

[the remark this answers is not in the document]
Comment (stumpylog) [reply src/documents/tests/test_date_parsing.py:95]: It's more there is already a `test_date_format_9` on line 103

**Review by stumpylog — APPROVED** [review]

**Review by qcasey — APPROVED** [review]

**Pull request merged**